### DISCUSSION TITLE
Bind/mirror custom regex capture group to behave like other non regex rule

### DISCUSSION THREAD

**Discussion by ssukienn** [renovatebot/renovate#27327] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Hi,

Could you guide me into direction where I would be able to somehow "bind" (or mirror the behaviour if such binding is impossible) custom regex group to other not regex rule?

My use case is that I have rule for `pnpm` bump like this:

```
    {
      matchDepNames: ['pnpm'],
      matchPackagePatterns: ['/pnpm$'],
      matchUpdateTypes: ['major'],
      addLabels: ['major'],
      groupName: 'pnpm major',
    },
```

which generally bump pnpm majors versions

and here is my generic custom regex matcher:
```
  customManagers: [
    {
      description: 'Regex manager for Dockerfile and github workflow files checking for *_VERSION with annotations',
      customType: 'regex',
      fileMatch: ['^Dockerfile.*$', '.github/workflows/.*\\.y[a]?ml$', ".*y[a]?ml$"],
      matchStrings: [
        'datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s+(ENV|ARGS)? ?\\w+?_VERSION[:=] ?(?<currentValue>.*)\\s',
      ],
      versioningTemplate: '{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}',
    },
  ],
```

which matches places like:
```
# renovate: datasource=npm depName=pnpm
ARG PNPM_VERSION=8.10.5
```

but matches any patch changes.

How could I change the regex rule (I am okay with seperating it to match only PNPM based versions if needed) so it only bumps major  versions. Or how to somehow bind that regex matcher to the rule mentioned first that matches depnames/update-type for `pnpm`. So basically, bump only major versions and at the same time.

Thank you for the guidance.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by secustor** [comment]
You need two separate packeRules. One which disables the undesired updates using `enabled: false` and a second one which matches whatever you want to a create a group for. 

Your first rule matching both depName and packageName does not make sense to me, you probably want to keep only one them.
